zone/beacon.cpp

```diff
@@ -121,16 +121,23 @@ bool Beacon::Process()
 
 void Beacon::AELocationSpell(Mob *caster, uint16 cast_spell_id, int16 resist_adjust)
 {
-	if(!IsValidSpell(cast_spell_id) || !caster)
+	if (!IsValidSpell(cast_spell_id) || !caster) {
 		return;
+	}
 
 	caster_id = caster->GetID();
 	spell_id = cast_spell_id;
 	this->resist_adjust = resist_adjust;
 	spell_iterations = spells[spell_id].aoe_duration / 2500;
 	spell_iterations = spell_iterations < 1 ? 1 : spell_iterations;	// at least 1
-	if (spells[spell_id].aoe_max_targets)
+	if (spells[spell_id].aoe_max_targets) {
 		max_targets = spells[spell_id].aoe_max_targets;
+	}
+
+	if (spells[cast_spell_id].target_type == ST_Ring) {
```

Comment:
I think I would rather switch `Beacon::Beacon` to take a position rather than a Mob and set the location that way (that's all it's used for in the ctor)

Reply:
And if you're wondering if we should take a vec3 or vec4, live always has their entities facing north :P (which means ones with push always push north on live, we currently don't have that issue)

Reply:
But it's so easy and it just works.